crates/uv-installer/src/site_packages.rs

```diff
@@ -47,11 +47,16 @@ impl<'a> SitePackages<'a> {
             // Read the site-packages directory.
             let site_packages = match fs::read_dir(site_packages) {
                 Ok(site_packages) => {
-                    let mut entries = site_packages.collect::<Result<Vec<_>, std::io::Error>>()?;
-                    // TODO(zanieb): Consider filtering to just directories to reduce the size of the sort
-                    // Sort for determinism, `read_dir` is different per-platform
-                    entries.sort_by_key(fs_err::DirEntry::path);
-                    entries
+                    // Use a `BTreeSet` to sort for determinism
+                    let directories: BTreeSet<_> = site_packages
+                        .filter_map(|read_dir| match read_dir {
+                            Ok(entry) => {
+                                entry.file_type().ok()?.is_dir().then_some(Ok(entry.path()))
```

Comment:
I think this is losing `file_type` errors because they're being cast to `None`.

Reply:
Ah yeah that is what I was having trouble avoiding.

cc @snowsignal

Reply:
```rust
let directories: BTreeSet<_> = site_packages
    .filter_map(|read_dir| match read_dir {
        Ok(entry) => {
            let file_type = match entry.file_type() {
                Ok(file_type) => file_type,
                Err(err) => return Some(Err(err)),
            };
            file_type.is_dir().then_some(Ok(entry.path()))
        }
        Err(err) => Some(Err(err)),
    })
    .collect::<Result<_, std::io::Error>>()?;
```

Reply:
Haha you beat me to it 😆

Reply:
Okay yuck though is this actually easier to read than before this pull request?

Reply:
I tried to clean it up a bit:

```rust
let directories: BTreeSet<_> = site_packages
    .filter_map(|read_dir| match read_dir {
        Ok(entry) => match entry.file_type() {
                Ok(file_type) => file_type.is_dir().then_some(Ok(entry.path())),
                Err(err) => Some(Err(err)),
            }
        Err(err) => Some(Err(err)),
    })
    .collect::<Result<_>>()?;
```